### PR TITLE
[REG-1304] Move generated scripts under RegressionGames asset path

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Moves generated scripts from `Assets/RGScripts` to `Assets/RegressionGames/Runtime/GeneratedScripts`
  - Handles removing the old folder to avoid classname conflicts
- Fixes an issue where generating scripts more than once would fail to generate the zip
- Updated path for the temp zip folder to avoid confusion with the `RegressionGames` folder in `Assets`

See also https://github.com/Regression-Games/RegressionDocs/pull/37

### CODE REVIEW

File: Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs [hunk 1]
```diff
@@ -16,6 +16,17 @@ public class RGActionCodeGenerator
...
+            // remove old 'RGScripts' folder that is no longer used
+            string dataPath = Application.dataPath;
+            string directoryToDelete = Path.Combine(dataPath, "RGScripts").Replace("\\", "/");
+
+            if (Directory.Exists(directoryToDelete))
+            {
+                Directory.Delete(directoryToDelete, true);
+                File.Delete(directoryToDelete + ".meta");
+                AssetDatabase.Refresh();
+            }
```
Comment (vontell) [inline Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs:28]: We can probably make a ticket to remove this at some point? I think we should get in the habit of cleaning up old cold that becomes outdated. Thoughts on that process?

**Review by vontell — APPROVED** [review]

File: Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs (hunk 1 above)
Comment (RG-nAmKcAz) [reply Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs:28]: I'm almost inclined to put a TODO here without a task # that lists a remove after X date indicator on it.  If someone is in there changing and sees it great.

I'm hesitant to suggest we create tasks for cleanup of single lines like this as historically all I've ever seen that do is add tasks to the backlog that never get prioritized :/

Comment (vontell) [reply Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs:28]: That's fine with me! I think a TODO is fine just so we know that this code can be removed in the future.

Comment (vontell) [reply Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs:28]: The other approach, if you wanted to do a ticket, is to make a ticket and explicitly put it into the cycle that is 2 cycles ahead from now, but I am fine with either way.

**Review by addisonbgross — APPROVED** [review]

File: Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs (hunk 1 above)
Comment (RG-nAmKcAz) [reply Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs:29]: todo added

**Pull request merged**